core/ledger/branch_manage.go

```diff
@@ -28,25 +27,35 @@ func (l *Ledger) GetBranchInfo(targetBlockid []byte, targetBlockHeight int64) ([
 	result := []string{}
 	it := l.baseDB.NewIteratorWithPrefix([]byte(pb.BranchInfoPrefix))
 	defer it.Release()
+
 	for it.Next() {
-		key := string(it.Key())
-		blockidStr := strings.Split(key, pb.BranchInfoPrefix)[1]
+		key := it.Key()
+		if len(key) < len(pb.BranchInfoPrefix)+1 {
+			// 理论上不会出现这种情况
+			continue
+		}
+
+		// key格式为:前缀+blockid，去掉前缀
+		blkId := key[len(pb.BranchInfoPrefix):]
```

Comment:
Do we need check if the string has the given prefix?

Reply:
it := l.baseDB.NewIteratorWithPrefix([]byte(pb.BranchInfoPrefix))
it.Next()
key := it.Key()

Key is found by prefix.